hal/src/common/thumbv6m/sercom/v2/spi.rs

```diff
@@ -1428,6 +1444,22 @@ impl<C: ValidConfig> Spi<C> {
         &self.config.as_ref().sercom()
     }
 
+    /// Update the SPI bus baudrate.
+    ///
+    /// **WARNING** this method assumes that the GCLK clock hasn't been changed
+    /// since [`Spi`] has been initialized.
```

Comment:
Can you change the clock after it has been used by a peripheral? I thought the ownership system would prevent such a thing?

Reply:
I think as it stands now, the `SercomCoreClock` isn't owned by the `Spi` struct, so it can be changed under an enabled `Spi`.

Reply:
The chips allow it. I'm not sure if the current `clock` module would allow it, but I intentionally didn't integrate with the existing `clock` module. I have a new `clock::v2` that is much more flexible. It can handle pretty much any clock configuration you want. The code is essentially done, but I need to find time to write the documentation. Part of that PR will be an update to `sercom::v2::spi` to integrate with it.